src/General-Rules-Tests/ReGuardClauseRuleTest.class.st

```diff
@@ -0,0 +1,50 @@
+"
+A ReGuardClauseRuleTest is a test class for testing the behavior of ReGuardClauseRule
+"
+Class {
+	#name : 'ReGuardClauseRuleTest',
+	#superclass : 'ReAbstractRuleTestCase',
+	#category : 'General-Rules-Tests-Coding Idiom Violation',
+	#package : 'General-Rules-Tests',
+	#tag : 'Coding Idiom Violation'
+}
+
+{ #category : 'tests' }
+ReGuardClauseRuleTest >> testRule [
+
+	self class compile: 'method | arg | arg := 1. arg > 2 ifTrue: [ 1. 1. ] ' classified: 'test-helper'.
+	[
+	| critiques |
+	critiques := self myCritiquesOnMethod: self class >> #method.
+	self assert: critiques size equals: 1 ] ensure: [ (self class >> #method) removeFromSystem ]
```

Comment:
The `removeFromSystem` action could be in a tearDown method, avoiding to use the `ensure:` block in every test.

Reply:
Indeed, not only for this class but also all rules tests. But I think it should be in another PR